### PR TITLE
Modify init CostBenefit()

### PR DESCRIPTION
**Pull request by zeliest**
Changes proposed in this PR:
- Update Init cost benefit

This PR fixes issue #

### PR Author Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] Correct target branch selected (if unsure, select `develop`)
- [ ] Source branch up-to-date with target branch
- [ ] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [ ] [Tests][testing] updated
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

### PR Reviewer Checklist

- [x] Read the [Contribution Guide][contrib]
- [x] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Reviewer_Checklist.html) passed
- [x] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: /CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Continuous_Integration_and_Testing.html
[linter]: https://climada-python.readthedocs.io/en/stable/guide/Guide_Continuous_Integration_and_Testing.html#3.C.--Static-Code-Analysis

### CODE REVIEW

File: climada/engine/cost_benefit.py [hunk 1]
```diff
@@ -134,28 +136,40 @@ class CostBenefit():
         'impact' (Impact): impact instance
     """
 
-    def __init__(self):
+    def __init__(self,
+                 present_year: Optional[int] = DEF_PRESENT_YEAR,
+                 future_year: Optional[int] = DEF_FUTURE_YEAR,
+                 tot_climate_risk: Optional[float] = 0.0,
+                 units:  Optional[str] = 'USD',
```
Comment (peanutfun) [inline climada/engine/cost_benefit.py:143]: Arguments with immutable values and defaults are not `Optional`:
```suggestion
                 present_year: int = DEF_PRESENT_YEAR,
                 future_year: int = DEF_FUTURE_YEAR,
                 tot_climate_risk: float = 0.0,
                 units: str = 'USD',
```

File: climada/engine/cost_benefit.py [hunk 2]
```diff
@@ -134,28 +136,40 @@ class CostBenefit():
...
         """Initilization"""
-        self.present_year = DEF_PRESENT_YEAR
-        self.future_year = DEF_FUTURE_YEAR
+        self.present_year = present_year
+        self.future_year = future_year
 
-        self.tot_climate_risk = 0.0
-        self.unit = 'USD'
+        self.tot_climate_risk = tot_climate_risk
+        self.unit = units
```
Comment (peanutfun) [inline climada/engine/cost_benefit.py:155]: Decide on singular or plural. Parameter should have the same name as attribute

File: climada/engine/cost_benefit.py [hunk 3]
```diff
@@ -134,28 +136,40 @@ class CostBenefit():
...
+    def __init__(self,
+                 present_year: Optional[int] = DEF_PRESENT_YEAR,
+                 future_year: Optional[int] = DEF_FUTURE_YEAR,
+                 tot_climate_risk: Optional[float] = 0.0,
+                 units:  Optional[str] = 'USD',
+                 color_rgb: Optional[dict] = None,
+                 benefit: Optional[dict] = None,
+                 cost_ben_ratio: Optional[dict] = None,
+                 imp_meas_present: Optional[dict] = None,
+                 imp_meas_future: Optional[dict] = None
```
Comment (peanutfun) [inline climada/engine/cost_benefit.py:148]: Possibly specify data types of dict keys and values using `typing.Dict`?

**Review by peanutfun — CHANGES_REQUESTED** [review]

File: climada/engine/cost_benefit.py (hunk 3 above)
Comment (zeliest) [reply climada/engine/cost_benefit.py]: I'm not sure how to do it for the imp_meas_present and imp_meas_future which are quite complex dictionaries

**Pull request merged**